src/frontend/src/components/public/PIAFormTabs/review/ProgamArea/displayProgramArea.tsx

```diff
@@ -0,0 +1,117 @@
+import ViewProgramAreaReview from '../viewProgramArea';
+import EditProgramAreaReview from '../editProgramArea';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+import { ApprovalRoles, PiaStatuses } from '../../../../../constant/constant';
+
+import { getGUID } from '../../../../../utils/helper.util';
+import messages from './../messages';
+import { IPiaForm } from '../../../../../types/interfaces/pia-form.interface';
+import { IReview } from '../interfaces';
+import { statusList } from '../../../../../utils/status';
+import {
+  IPiaFormContext,
+  PiaFormContext,
+} from '../../../../../contexts/PiaFormContext';
+import { useContext } from 'react';
+
+export interface IDisplayProgramAreaProps {
+  stateChangeHandler: (value: any, path: string, callApi?: boolean) => void;
+  pia: IPiaForm;
+  reviewForm: IReview;
+  mandatoryADM: boolean;
+}
+
+const DisplayProgramArea = (props: IDisplayProgramAreaProps) => {
+  const { pia, piaStateChangeHandler } =
+    useContext<IPiaFormContext>(PiaFormContext);
+  const allowUserReviewProgramArea = () => {
+    // if selectedRoles is null means none of selectedRole got reviewed so return true
+    // otherwise loop all the role in reviews part to see if the current user already did review
+
+    const userGuid = getGUID();
+    const selectedRoles = props.reviewForm?.programArea?.selectedRoles;
+    if (selectedRoles === null) return true;
+    for (const role of selectedRoles) {
+      if (
+        props.reviewForm?.programArea?.reviews?.[role]?.reviewedByGuid ===
+        userGuid
+      ) {
+        return false;
+      }
+    }
+
+    return true;
+  };
+  return (
+    <div>
+      {statusList(null)?.[Object(props.pia).status]?.Pages?.review?.params
+        ?.editProgramArea && <h4 className="mb-3">Selected Roles</h4>}
+      {props.reviewForm.programArea?.selectedRoles.length > 0 ? (
+        props.reviewForm.programArea?.selectedRoles.map(
```

Comment:
For readability it would be nice to see a `const selectedRoles = props.reviewForm.programArea?.selectedRoles;